lsp/nls/src/field_walker.rs

```diff
@@ -128,6 +141,11 @@ pub struct DefWithPath {
     pub value: Option<RichTerm>,
     /// The path within the value that this binding refers to.
     pub path: Vec<Ident>,
+    /// If this definition came from a record field, this is the containing
+    /// record. Note that having a parent record is mutually exclusive with
+    /// having a non-empty path, because non-empty paths only come from let
+    /// pattern bindings.
```

Comment:
Would that make sense to encode this invariant as a type, instead of a comment (a simple enum with maybe some helper functions)?